src/engine.ml

```diff
@@ -104,22 +104,24 @@ let compute_hmac key p hmac_algorithm hmac_key =
 
 let hmac_and_out protocol { hmac_algorithm; my_hmac; _ } key
     (p : [< Packet.ack | Packet.control ]) =
-  let hmac = compute_hmac key p hmac_algorithm my_hmac in
-  let header = Packet.header p in
-  let p' = Packet.with_header { header with Packet.hmac } p in
-  Packet.encode protocol (key, p')
+  (* A dummy hmac. Unfortunately, it has to be the correct length for now *)
```

Comment:
I don't quite understand this comment...

Reply:
Ah, it's no longer relevant since c635ce5576677614595ee3bc9343d5b426ed66e0. Before that a `Cstruct.create hmac_len` was added in the `p` header.

Reply:
To expand a bit more: `Packet.encode` figured out the hmac length from the `hmac` cstruct member of the header. This design required passing a dummy hmac value of the correct length which was wasteful. Then I refactored so the int `hmac_len` is passed to `Packet.encode` and the `hmac` member is ignored (=> we can use `Cstruct.empty`!). Then the correct hmac is set using `Packet.set_hmac`.